contracts/orderbook/Keepers/OrderKeeper.sol

```diff
@@ -0,0 +1,48 @@
+pragma solidity ^0.8.0;
+import "../IOrderBook.sol";
+import "@openzeppelin-4.3.2/token/ERC20/extensions/IERC20Metadata.sol";
+import "../../governance/PausableGuardian_0_8.sol";
+
+contract OrderKeeper is PausableGuardian_0_8 {
+    IOrderBook public factory;
+
+    constructor(IOrderBook factoryAddress) {
+        factory = factoryAddress;
+    }
+
+    function checkUpKeep(bytes calldata checkData)
```

Comment:
so this upkeep is can hanlde one order at a time. which is not really efficient. we definitely can bundle multiple orders in single tx

Reply:
not exactly unfortunately. If the orders are for the same pair what happens is executing one might mess up execution of the other with the price so it wont be executable. Also the gas for 2 orders might push over 2.5 mln limit as 1 is at 1 mln w/o uni v3 (depending on swap complexity it can be above 1.3 mln gas)